google-cloud-firestore/src/test/java/com/google/cloud/firestore/LocalFirestoreHelper.java

```diff
@@ -970,18 +978,88 @@ public static String autoId() {
   }
 
   private static Map<String, Object> fromJsonString(String json) {
-    try {
-      ObjectMapper mapper = new ObjectMapper();
-      return mapper.readValue(json, new TypeReference<Map<String, Object>>() {});
-    } catch (IOException e) {
-      throw new RuntimeException(e);
-    }
+    Type type = new TypeToken<Map<String, Object>>() {}.getType();
+    GsonBuilder gsonBuilder = new GsonBuilder();
+    gsonBuilder.registerTypeAdapterFactory(CustomObjectTypeAdapter.FACTORY);
+    Gson gson = gsonBuilder.create();
+    return gson.fromJson(json, type);
   }
 
   public static Map<String, Object> fromSingleQuotedString(String json) {
     return fromJsonString(json.replace("'", "\""));
   }
 
+  static class CustomObjectTypeAdapter extends TypeAdapter<Object> {
```

Comment:
Needed to write this Custom TypeAdapter because GSON considers every numeric value to be Double by deafult and there is no neat way around it.

Reply:
In javaScript and JSON all numbvers are doubles. There are no ints.